src/starterbot/BaseManager.cpp

```diff
@@ -0,0 +1,156 @@
+#include "BaseManager.h"
+
+BaseManager::BaseManager(UnitManager& unitManager) :
+    m_unitManager(unitManager) {
+}
+
+Base& BaseManager::getClosestBase(bw::Unit& unit) {
+    Base* closestBase;
+    int shortestDistance = INT_MAX;
+
+    // getDistance uses Brood War's built-in pathfinding, so this will find the
+    // base that's closest while accounting for terrain.
```

Comment:
This is not correct; `UnitInterface::getDistance()` just calls `Position::getApproxDistance()` under the hood. So it gets the distance as the crow flies rather than the ground distance with terrain.

I don't think BWAPI has any functions for pathfinding distance, but I've found `getDistance()` to be a good enough approximation as it is.